### ISSUE TITLE
Dependencies for string overwriting don't seem to work

### ISSUE THREAD

**Issue by juergmessmer** [rapila/cms-base#206]
If I make a placeholder string in a widget, that can to be overwritten by another plugin or for a special site, I expect that a string (key) added to a plugin2 - which depends on plugin1 - is displayed instead of the original one. But that doesn't seem to work - even if I use the same path / dir structure in both plugins.

Plugin1 
info.yml
dependencies:
  plugins/plugin2: "*"
placeholder.string-key: ""

Plugin2
placeholder.string-key: "My overwrite string"

**Comment by sabberworm** [comment]
If you add the plugin 2 to the list of dependencies of plugin 1, that means plugin 1 depends on plugin 2 so plugin 2 can override plugin 1 but not the other way around. The behaviour you're seeing is thus correct and you either need to switch around your strings or the dependency direction do that plugin 1 becomes a dependee of plugin 2.

**Comment by sabberworm** [comment]
Think of it this way: site depends on base because site can’t function without base whereas the reverse is (somewhat) true. That’s why site can override/extend base but not the other way around.

**Comment by juergmessmer** [comment]
Ok, thanks. I solved it by using {{writeString=placeholder.string-key;defaultValue=""}} in the template instead of a ini string. Now it works!

**Comment by sabberworm** [comment]
Which plugins are were you talking about?

**Comment by juergmessmer** [comment]
Wettingen overwriting SchulCMS. I wanted to define guidelines for the class portrait. But this is only Wettingen specific.

**Comment by sabberworm** [comment]
Yes but that should work. Wettingen has plugins/schulcms in its dependencies.

**Comment by juergmessmer** [comment]
But it didn't. Now it does.

**Comment by sabberworm** [comment]
Reopening this as it seems to be a real issue (despite the faulty description).

**Comment by juergmessmer** [comment]
Ok, I corrected the previously wrong (confused Plugin1/2) 
or shouldn't this be done at this stage?

**Comment by sabberworm** [comment]
My tests show the following:

`plugins/«p»/modules/«t»/«m»/lang/«l».ini` can be overridden by `site/modules/«t»/«m»/lang/«l».ini` but not by `site/lang/«l».ini`. I don’t know if that is supposed to be correct…

**Comment by juergmessmer** [comment]
I havn't had any problems with this anymore. I guess we close it for now?
